bin/tests/check_dl_core_wrapper_callback.sh

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env bash
+
+# Cause the script to exit if a single command fails
+set -eo pipefail -v
+
+
+################################  global variables  ################################
+rm -rf ./tests/logs ./tests/output.txt
+
+EXPDIR=./tests/_tests_dl_callbacks
+LOGDIR=./tests/logs/_tests_dl_callbacks
+CHECKPOINTS=${LOGDIR}/checkpoints
+LOGFILE=${CHECKPOINTS}/_metrics.json
+EXP_OUTPUT=./tests/output.txt
+
+
+function check_file_existence {
+    # $1 - path to file
+    if [[ ! -f "$1" ]]
+    then
+        echo "There is no '$1'!"
+        exit 1
+    fi
+}
+
+
+function check_num_files {
+    # $1 - ls directory
+    # $2 - expected count
+    NFILES=$( ls $1 | wc -l )
+    if [[ $NFILES -ne $2 ]]
+    then
+        echo "Different number of files in '$1' - "`
+              `"expected $2 but actual number is $NFILES!"
+        exit 1
+    fi
+}
+
+
+function check_checkpoints {
+    # $1 - file prefix
+    # $2 - expected count
+    check_num_files "${1}.pth" $2
+    check_num_files "${1}_full.pth" $2
+}
+
+
+function check_line_counts {
+    # $1 file
+    # $2 pattern
+    # $3 expected count
+    ACTUAL_COUNT=$( grep -c "$2" $1 || true )  # '|| true' for handling pipefail
+    if [ $ACTUAL_COUNT -ne $3 ]
+    then
+        echo "Different number of lines in file '$1' - "`
+             `"expected $3 (should match '$2') but actual number is $ACTUAL_COUNT!"
+        exit 1
+    fi
+}
+
+# ################################  pipeline 00  ################################
+# setup: ignore loss when it is not a main metric
+LOG_MSG='pipeline 00'
+echo ${LOG_MSG}
+
+LOGDIR=./tests/logs/_tests_dl_callbacks
+CHECKPOINTS=${LOGDIR}/checkpoints
+LOGFILE=${CHECKPOINTS}/_metrics.json
+EXP_OUTPUT=./tests/output.txt
+
+PYTHONPATH=./examples:./catalyst:${PYTHONPATH} \
+  python3 -c "
```

Comment:
btw, could you please move your tests to .py files,
like https://github.com/catalyst-team/catalyst/blob/master/catalyst/contrib/dl/callbacks/tests/test_tracer_callback.py
?